lib/gwdb-legacy/database.ml

```diff
@@ -326,25 +343,34 @@ let new_persons_of_first_name_or_surname cmp_str cmp_istr base_data params =
         let ic_dat = Secure.open_in_bin fname_dat in
         seek_in ic_dat pos;
         let len = input_binary_int ic_dat in
-        let rec read_loop ipera len =
-          if len = 0 then ipera
+        let rec read_loop (ipera, iperset) len =
+          if len = 0 then (ipera, iperset)
           else
             let iper = input_binary_int ic_dat in
-            read_loop (iper :: ipera) (len - 1)
+            let iperset' = Ext_int.Set.add iper iperset in
+            let ipera = if iperset' != iperset then iper :: ipera else ipera in
+            read_loop (ipera, iperset') (len - 1)
         in
-        let ipera = read_loop [] len in
+        let ipera, iperset = read_loop ([], Ext_int.Set.empty) len in
         close_in ic_dat;
-        ipera
-      with Not_found -> []
+        (ipera, iperset)
+      with Not_found -> ([], Ext_int.Set.empty)
     in
-    let patched = Hashtbl.fold (fun i _ acc -> i :: acc) person_patches [] in
-    let ipera = List.filter (fun i -> not @@ List.mem i patched) ipera in
+    let iperset = Ext_int.Set.diff iperset (Lazy.force patched_set) in
+    let ipera = List.filter (fun iper -> Ext_int.Set.mem iper iperset) ipera in
```

Comment:
```suggestion
    let ipera = List.filter (fun iper -> Ext_int.Set.mem iper (Lazy.force patched_set)) ipera in
```

No?

Reply:
could be
`let ipera = List.filter (fun iper -> not (Ext_int.Set.mem iper (Lazy.force patched_set))) ipera in `

Reply:
Sorry, just ignore my comment.